src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MonthCalendarTests.cs

```diff
@@ -4505,6 +4505,225 @@ public void MonthCalendar_HitTest_ReturnsExpected(int x, int y, HitArea expected
+        ShouldSerializeProperty(() => calendar.TestAccessor().Dynamic.ShouldSerializeMaxDate(), () => calendar.MaxDate = DateTime.Now.AddYears(2));
+        ShouldSerializeProperty(() => calendar.TestAccessor().Dynamic.ShouldSerializeMinDate(), () => calendar.MinDate = DateTime.Now);
+        ShouldSerializeProperty(() => calendar.TestAccessor().Dynamic.ShouldSerializeTrailingForeColor(), () => calendar.TrailingForeColor = Color.Red);
+        ShouldSerializeProperty(() => calendar.TestAccessor().Dynamic.ShouldSerializeTitleForeColor(), () => calendar.TitleForeColor = Color.Red);
+        ShouldSerializeProperty(() => calendar.TestAccessor().Dynamic.ShouldSerializeTitleBackColor(), () => calendar.TitleBackColor = Color.Red);
+    }
+
+    private void ShouldSerializeProperty(Func<bool> shouldSerializeFunc, Action setPropertyAction)
+    {
+        bool result1 = shouldSerializeFunc();
+        result1.Should().BeFalse();
+
+        setPropertyAction();
+
+        bool result2 = shouldSerializeFunc();
+        result2.Should().BeTrue();
+    }
+
+    [WinFormsTheory]
+    [BoolData]
+    public void MonthCalendar_ResetTodayDate_InvokeWithSetTodayDate_ReturnsExpected(bool createHandle)
+    {
+        using MonthCalendar calendar = new();
+        calendar.TodayDate = new DateTime(2000, 1, 1);
+
+        if (createHandle)
+        {
+            nint handle = calendar.Handle;
+        }
+
+        DateTime todayDateBeforeReset = calendar.TodayDate;
+        calendar.TestAccessor().Dynamic.ResetTodayDate();
+        DateTime todayDateAfterReset = calendar.TodayDate;
+
+        todayDateBeforeReset.Should().NotBe(todayDateAfterReset);
+        todayDateAfterReset.Should().Be(DateTime.Today);
+    }
+
+    [WinFormsTheory]
+    [BoolData]
+    public void MonthCalendar_ResetTodayDate_InvokeWithoutSetTodayDate_ReturnsExpected(bool createHandle)
+    {
+        using MonthCalendar calendar = new();
+
+        if (createHandle)
+        {
+            nint handle = calendar.Handle;
+        }
+
+        DateTime todayDateBeforeReset = calendar.TodayDate;
+        calendar.TestAccessor().Dynamic.ResetTodayDate();
+        DateTime todayDateAfterReset = calendar.TodayDate;
+
+        todayDateBeforeReset.Should().Be(todayDateAfterReset);
+    }
+
+    private void TestResetColorProperty(Func<MonthCalendar, Color> getColor, Action<MonthCalendar, Color> setColor, Action<dynamic> resetColor)
+    {
+        using MonthCalendar calendar = new();
+        var testAccessor = calendar.TestAccessor();
+
+        Color originalColor = getColor(calendar);
+
+        setColor(calendar, Color.Red);
+        getColor(calendar).Should().Be(Color.Red);
+
+        resetColor(testAccessor.Dynamic);
+        getColor(calendar).Should().Be(originalColor);
+
+        setColor(calendar, Color.Red);
+        calendar.CreateControl();
+        _ = calendar.Handle;
```

Comment:
when we access `Handle`, if the handle has not yet been created, this (`_ = calendar.Handle`)  will force handle creation.  
It's not recommended to simply write `calendar.Handle`, because that can be confusing, we don't konw if you are trying to **access** `Handle` or to **create** `Handle`.

so in this case, if you want to create handle here, `calendar.CreateControl();` will create handle, so  `calendar.Handle` is not necessary.

```suggestion
        calendar.CreateControl();
```